src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpTypingStyleDiagnosticAnalyzerBase.cs

```diff
@@ -0,0 +1,272 @@
+                             : TypingStyles.NoVarWhereApparent;
+
+            stylePreferences |= optionSet.GetOption(CSharpCodeStyleOptions.UseVarWherePossible)
+                             ? TypingStyles.VarWherePossible
+                             : TypingStyles.NoVarWherePossible;
+
+            return stylePreferences;
+        }
+
+        protected bool IsTypeApparentInDeclaration(SyntaxNode declarationStatement, SemanticModel semanticModel, TypingStyles stylePreferences, CancellationToken cancellationToken)
+        {
+            // use var in foreach statement to make it concise.
+            if (declarationStatement.IsKind(SyntaxKind.ForEachStatement))
+            {
+                return true;
+            }
+
+            // variable declaration cases.
+            var variableDeclaration = (VariableDeclarationSyntax)declarationStatement;
+            var initializer = variableDeclaration.Variables.Single().Initializer;
+            var initializerExpression = GetInitializerExpression(initializer);
+
+            // default(type)
+            if (initializerExpression.IsKind(SyntaxKind.DefaultExpression))
+            {
+                return true;
+            }
+
+            // literals, use var if options allow usage here.
+            if (initializerExpression.IsAnyLiteralExpression())
+            {
+                return stylePreferences.HasFlag(TypingStyles.VarForIntrinsic);
+            }
+
+            // constructor invocations cases:
+            //      = new type();
+            if (initializerExpression.IsKind(SyntaxKind.ObjectCreationExpression) &&
+                !initializerExpression.IsKind(SyntaxKind.AnonymousObjectCreationExpression))
+            {
+                return true;
+            }
+
+            // explicit conversion cases: 
+            //      (type)expr, expr is type, expr as type
+            if (initializerExpression.IsKind(SyntaxKind.CastExpression) ||
+                initializerExpression.IsKind(SyntaxKind.IsExpression) ||
+                initializerExpression.IsKind(SyntaxKind.AsExpression))
+            {
+                return true;
+            }
+
+            // other Conversion cases:
+            //      a. conversion with helpers like: int.Parse, TextSpan.From methods 
+            //      b. types that implement IConvertible and then invoking .ToType()
+            //      c. System.Convert.Totype()
+            var declaredTypeSymbol = semanticModel.GetTypeInfo(variableDeclaration.Type, cancellationToken).Type;
+            var expressionOnRightSide = initializerExpression.WalkDownParentheses();
+
+            var memberName = expressionOnRightSide.GetRightmostName();
+            if (memberName == null)
+            {
+                return false;
+            }
+
+            var possibleMethodSymbol = semanticModel.GetSymbolInfo(memberName, cancellationToken).Symbol;
+            if (possibleMethodSymbol == null || !possibleMethodSymbol.IsKind(SymbolKind.Method))
```

Comment:
You could probably clean this up a bit and combine the next variable declaration by doing something like...

```C#
var methodSymbol = semanticModel.GetSymbolInfo(memberName, cancellationToken).Symbol as IMethodSymbol;
if (methodSymbol == null)
{
    return false;
}
```